Automattic/jetpack
# My Jetpack: Add "Expired" & "Expires soon" statuses to product cards

This PR adds an "Expired" or "Expiring soon" status to the product cards of products with an expired or expiring paid plan subscription. We also make the product CTA say "Renew my plan" (or "Resume my plan" if expired) and link to the specific subscription in purchase management section of their wordpress.com account so that they can renew or resume their subscription.





### Screenshots:

![Screen Shot 2024-11-26 at 08 12 58](https://github.com/user-attachments/assets/090e4024-a67d-4e56-9143-261a4b0b4d6d)

![Screen Shot 2024-10-22 at 18 57 57](https://github.com/user-attachments/assets/6131c4f6-df0f-40a2-aeb0-4e02cf5c05f5)

.

## Proposed changes:

* Added an "Expired" or "Expiring soon" status to the My Jetpack product cards.
* For products that are Expired, the CTA says "Resume my plan" and points to the wordpress.com purchase management page. For products that are Expiring soon, the CTA says "Renew my plan", and points directly to checkout with the product renewal in the cart.
* Also fixes issue: [Backup Card: My Jetpack shows "Needs Plan" for backup even when there is a Jetpack plan from a hosting partner](https://github.com/Automattic/jetpack/issues/39762). - Testing instructions below ⤵️

### Other information:


## Jetpack product discussion



Project Thread: pbNhbs-bJN-p2

Design Thread: p1HpG7-umC-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:





**To test this PR, skip down to "Testing this PR":**
Otherwise, if testing issue: _Backup Card: My Jetpack shows "Needs Plan" for backup even when there is a Jetpack plan from a hosting partner_, follow the instructions below:


#### Testing issue https://github.com/Automattic/jetpack/issues/39762 :<br/>(Backup Card: My Jetpack shows "Needs Plan" for backup even when there is a Jetpack plan from a hosting partner.)
- Create a Pressable hosting partner site (if you don't already have one) by following instructions here: PCYsg-9TF-p2
    - (sign up using the direct link to the a8c employee plan that requires no coupon)
- Install the Jetpack Beta Tester plugin: https://github.com/Automattic/jetpack-beta
- Activate this PR branch for Jetpack in Jetpack Beta plugin.
- Purchase a Backup plan.
- Open My Jetpack and confirm that Backup product card status is Active and not "Needs plan". 

#### Testing this PR:
* Test this PR with a Jurassic Ninja Site running the Beta plugin on the `add/mj-expired-products-cards` branch. In the JN interface, you can automatically assign paid product licenses for all the products shown on the My Jetpack page, See screenshot:

![Screen Shot on 2024-10-22 at 22-08-12](https://github.com/user-attachments/assets/e041055f-71fd-4650-b46a-8f79025ea6f6)
* Connect Jetpack, both site and user.  (Once user connected, you will see all the paid plans in My Jetpack.)
* Confirm that all paid products are showing the correct status (For example confirm Backup and Search are not showing "Needs plan" when the paid plan has actually been purchased) - See issue: https://github.com/Automattic/jetpack/issues/39762
* Now look up your site in Store Admin. 
* One by one, for each of the subscriptions, change the expiration date to either expiring (nearing expiration), or expired . You may need to disable 'auto-renew' for the plan to show as expiring/expired. The go to My Jetpack, reload the page and look at the product card for that plan.
* Verify the status is either "Expired" or "Expiring soon", depending on the date you set.
* Verify there is a red border around expired products, and a yellow border around products expiring soon.
* Verify that for products that are Expired, the CTA says "Resume my plan" and points to the wordpress.com purchase management page. For products that are Expiring soon, verify the CTA says "Renew my plan", and points directly to checkout with the product renewal in the cart.
* Continue this (set the expiration date, then view in My Jetpack) for each of the products.
* Once done testing all products, in Store Admin you can remove all the product subscriptions.
* Now Purchase a bundle for the site, such as Complete, or Security.
* In Store Admin, set the bundle subscription date to expired/expiring. (you may need to disable auto-renew).
* Now view the products included with the bundle on My Jetpack.  The products included with the bundle should be showing as expired/expiring soon, in My Jetpack.
* Verify that the expired/expiring CTA points to the bundle subscription in the purchase management page on wordpress.com (for expired), or points directly to checkout (for expiring-soon).
* Establish a site-only connection (no user-connection) and verify everything works properly with no errors. Some product cars may now show the status "Requires user connection", which takes precedence over expired/expiring products.
